src/mrsimulator/spin_system/isotope.py

```diff
@@ -42,57 +51,80 @@ class Isotope(BaseModel):
     """
 
     symbol: str
+    spin_multiplicity: int = Field(default=2, ge=2)
+    gyromagnetic_ratio: float = 0
+    quadrupole_moment: float = 0
+    natural_abundance: float = Field(default=100, ge=0, le=100)
+    atomic_number: int = Field(default=0)
+
     test_vars: ClassVar[Dict] = {"symbol": "1H"}
+    custom_isotope_data: ClassVar[Dict] = {}
 
     class Config:
         extra = "forbid"
         validate_assignment = True
+        allow_mutation = False
+
+    def __init__(self, **kwargs):
+        symbol = kwargs["symbol"]
+        if symbol not in get_all_isotope_symbols():
+            raise Exception(f"Isotope symbol `{symbol}` not recognized.")
+        kwargs_new = get_isotope_dict(symbol)
+        for k, v in kwargs.items():
+            if v != kwargs_new[k]:
+                raise ValueError(f"{k} for {symbol} cannot be assigned.")
+        super().__init__(**kwargs_new)
+
+    @classmethod
+    def register(cls, symbol, copy_from=None, **kwargs):
+        if symbol in ISOTOPE_DATA.keys():
```

Comment:
This only checks if the passed symbol exists in `ISOTOPE_DATA` (i.e. is a real isotope). Calling register again with the same symbol will mutate the stored attributes. For example:

```
>>> Isotope.register(symbol="x")
>>> Isotope(symbol="x").spin_multiplicity  # Default
2
>>> Isotope.register(symbol="x", spin_multiplicity=5)
>>> Isotope(symbol="x").spin_multiplicity
5
```

`allow_mutation` is set to `False` in the config for the class, but this only applies to instances of the `Isotope` class. @deepanshs do we want to prevent mutation of registered isotope information as well?

Reply:
I don't think we want to restrict the redefinition of custom isotopes. Calling the register function twice will update the info of `x`, and it's fine. It's like updating the value of a variable.